programs/stake/src/stake_state.rs

```diff
@@ -431,6 +431,13 @@ impl Delegation {
             None
         }
     }
+
+    fn can_merge_with(&self, source: &Self) -> bool {
+        self.voter_pubkey == source.voter_pubkey
```

Comment:
similar to this my confusion (https://github.com/solana-labs/solana/pull/13712/files#r528207212), `voter_pubkey` check is only needed for fully active stakes, not for fully-deactivated stakes? And, that's reason the old-/new- code behavior for deactivated stakes doesn't check `voter_pubkey`, right? It might be helpful for  this subtlety to be expressed in some way as a code structure or any commenting. :)

Reply:
I find it easier to think of the `Stake` struct as invalid after cooldown.  Aside from determining that the account is deactivated, its fields will never be referenced again.  They will be entirely overwritten should the account be redelegated in the future.